src/protocol/produce.rs

```diff
@@ -24,19 +27,22 @@ pub struct ProduceRequest<'a, 'b> {
     pub timeout: i32,
     pub topic_partitions: Vec<TopicPartitionProduceRequest<'b>>,
     pub compression: Compression,
+    pub api_version: Rc<ApiVersion>,
```

Comment:
1. Yes, there is a `api_version` field in the request header, but `kafka-rust` hard code it to 0. So, we have to add a real `ApiVersion` to decide whether the server use the wrong frame checksum (0.8/0.9) or not (0.10 later).